### PR TITLE
Support streaming datasets with pyarrow.parquet.read_table

### PR DESCRIPTION
**Pull request by albertvillanova**
Support streaming datasets with `pyarrow.parquet.read_table`.

See: https://huggingface.co/datasets/uonlp/CulturaX/discussions/2

CC: @AndreaFrancis

### CODE REVIEW

**Comment by mariosasko** [comment]
This function reads an entire Arrow table in one go, which is not ideal memory-wise, so I don't think we should encourage using this function, considering we want to keep RAM usage as low as possible in the streaming mode. 

(Note that Parquet files are compressed, meaning the loaded table can be significantly larger than the size in Parquet.)

Instead, we should suggest the authors to use:
```python
with open(doc_path, "rb") as f:
    parquet_file = pq.ParquetFile(f)
    for batch in parquet_file.iter_batches():
        pa_table = pa.Table.from_batches([batch])
        yield idx, pa_table
        idx += 1
```

**Comment by albertvillanova** [comment]
@mariosasko I think the potential problem you evoke is independent of whether or not we support streaming mode:
- if the user's script with `read_table` works in non-streaming mode, it will also work in streaming mode after this PR

In fact, what we should suggest instead is to follow the scriptless approach, so that our `parquet` packaged module is used, with all the optimizations implemented. But this approach is not possible in all cases, and some use cases need to implement a script. And if they have small Parquet files and use `read_table`, I think we should support streaming.

In summary, let me clarify the goal and the scope of this PR:
- a user needs using a loading script
- their files are small enough so that they use `read_table`
- their loading script works in non-streaming mode
- therefore, this PR allows loading their dataset in streaming mode as well

**Comment by mariosasko** [comment]
Yes, the no-script approach with metadata configs makes the most sense.

> their files are small enough so that they use read_table

Some of the Parquet files in that repo are larger than 1 GB ...

Also, I'd wait for more instances of people using the `read_table` function on the Hub before merging this PR.

**Comment by albertvillanova** [comment]
@mariosasko, yes, this solution is not specifically for the "uonlp/CulturaX" dataset, but for other use cases as I explained above: indeed, they finally removed the use of `read_table` because their data files are too large.

> Also, I'd wait for more instances of people using the `read_table` function on the Hub before merging this PR.

Do you know how many datasets are currently using `read_table`?

**Comment by mariosasko** [comment]
> Do you know how many datasets are currently using read_table?

Zero (based on the script that checks the script contents of the public Hub datasets).

**Comment by albertvillanova** [comment]
I see... Thanks! :hugs:

**Comment by albertvillanova** [comment]
I'm merging this PR as discussed in private.
